### PR TITLE
[4.0] Group action buttons for com_modules

### PR DESCRIPTION
**Pull request by bembelimen**
### Summary of Changes
This PR continues the grouping of action buttons in the backend (see https://github.com/joomla/joomla-cms/pull/22832, https://github.com/joomla/joomla-cms/pull/22393, https://github.com/joomla/joomla-cms/pull/23337 and https://github.com/joomla/joomla-cms/pull/23336). This time the modules buttons.

### Testing Instructions
Go to the modules manager in the backend and check the toolbar. There should be a "Change status" dropdown now which activates when you select a module item.

### CODE REVIEW

**Comment by ChristineWk** [comment]
I have tested this item :white_check_mark: successfully on 4fae8e08cfacd5e42dc52262c4b7ab46a7ddebd3<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/23378">issues.joomla.org/tracker/joomla-cms/23378</a>.</sub>

File: administrator/components/com_modules/View/Modules/HtmlView.php [hunk 1]
```diff
@@ -161,52 +161,68 @@ protected function addToolbar()
 
 		if ($canDo->get('core.create'))
 		{
-			// Instantiate a new FileLayout instance and render the layout
-			$layout = new FileLayout('toolbar.newmodule');
+			$toolbar->standardButton('new', 'JTOOLBAR_NEW')
+					->onclick("location.href='index.php?option=com_modules&amp;view=select'");
```
Comment (Quy) [inline administrator/components/com_modules/View/Modules/HtmlView.php]:
```suggestion
				->onclick("location.href='index.php?option=com_modules&amp;view=select'");
```

**Comment by Quy** [comment]
Add the deleted file to the list in `administrator/components/com_admin/script.php`?

File: administrator/components/com_modules/View/Modules/HtmlView.php (hunk 1 above)
Comment (bembelimen) [reply administrator/components/com_modules/View/Modules/HtmlView.php]: Thx

**Comment by Hackwar** [comment]
I have tested this item :white_check_mark: successfully on f38ac6998efda5fccf2c9bb754e1ca1b0d4f8d81<br /><br />Works as expected. As was said in another PR, we don't need to add files to the delete list, since that one will be automatically generated from script shortly before probably RC release.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/23378">issues.joomla.org/tracker/joomla-cms/23378</a>.</sub>

**Comment by PhocaCz** [comment]
Tested successfully in 4.0.0-alpha6-dev :ok: :heavy_check_mark: 

OK

File: administrator/components/com_modules/View/Modules/HtmlView.php [hunk 2]
```diff
@@ -161,52 +161,68 @@ protected function addToolbar()
...
 
-			$bar->appendButton('Custom', $layout->render(array()), 'new');
+			$toolbar->standardButton('copy')
+				->text('JTOOLBAR_DUPLICATE')
+				->task('modules.duplicate')
+				->listCheck(true);
 		}
 
-		if ($canDo->get('core.create'))
+		if ($canDo->get('core.edit.state') || Factory::getUser()->authorise('core.admin'))
```
Comment (Quy) [inline administrator/components/com_modules/View/Modules/HtmlView.php:173]:
```suggestion
		if ($canDo->get('core.edit.state') || $user->authorise('core.admin'))
```

File: administrator/components/com_modules/View/Modules/HtmlView.php [hunk 3]
```diff
@@ -161,52 +161,68 @@ protected function addToolbar()
...
+			$childBar = $dropdown->getChildToolbar();
+
+			if ($canDo->get('core.edit.state'))
+			{
+				$childBar->publish('modules.publish')->listCheck(true);
+
+				$childBar->unpublish('modules.unpublish')->listCheck(true);
+			}
+
+			if (Factory::getUser()->authorise('core.admin'))
```
Comment (Quy) [inline administrator/components/com_modules/View/Modules/HtmlView.php:191]:
```suggestion
			if ($user->authorise('core.admin'))
```

**Comment by drmenzelit** [comment]
I have tested this item :white_check_mark: successfully on f38ac6998efda5fccf2c9bb754e1ca1b0d4f8d81<br /><br />Tested succesfully on Joomla! 4.0.0-alpha7-dev<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/23378">issues.joomla.org/tracker/joomla-cms/23378</a>.</sub>

**Comment by PhocaCz** [comment]
I have tested this item :white_check_mark: successfully on f38ac6998efda5fccf2c9bb754e1ca1b0d4f8d81<br /><br />Tested now in Joomla! Issue Tracker - CMS

Tested succesfully on Joomla! 4.0.0-alpha7-dev<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/23378">issues.joomla.org/tracker/joomla-cms/23378</a>.</sub>

**Comment by wilsonge** [comment]
Thanks!

**Pull request merged**